src/contracts/core/StakeRootCompendium.sol

```diff
@@ -538,15 +594,25 @@ contract StakeRootCompendium is StakeRootCompendiumStorage {
                 operatorTreeRoot,
                 keccak256(
                     abi.encodePacked(
-                        totalDelegatedStake, 
-                        totalSlashableStake, 
+                        totalDelegatedStake,
+                        totalSlashableStake,
                         operatorSetExtraDatas[operatorSet.avs][operatorSet.operatorSetId]
                     )
                 )
             )
         );
     }
 
+    function safeTransferETH(address to, uint256 amount) internal {
```

Comment:
assembly negatively impacts readability and review time. should only be used when necessary or in the hot path (proof verification) iff gas savings are significant.

i'd remove this or if we decide to use this pattern else where stick it in a library somewhere

Reply:
This is more of a codesize optmization than it is an execution one, I do like the idea of it having it's own library tho.

Reply:
looks like it saves <100 bytes of contract size?

I think we can move the view functions to generate the stake root out of the compendium if the contract gets too large.

StakeRootManager (billing, operatorSet registration, stakeRoot submission), StakeRoot (view functions referencing the StakeRootManager state.

Reply:
can we leave it as 

`        (bool success, ) = to.call{value: amount}("");
        if (!success) {
            revert EthTransferFailed();
        }`

for readability?

Reply:
to be fair both are pretty unreadable and we're using the logic in multiple places, a wrapper should be used at the very least imo.

Reply:
yeah i mean keep the internal function but replace the assembly with solidity